Sources/TensorFlow/Operators/NN.swift

```diff
@@ -142,46 +142,56 @@ func _vjpConv2D<Scalar: TensorFlowFloatingPoint>(
 ) -> (value: Tensor<Scalar>, pullback: (Tensor<Scalar>) -> (Tensor<Scalar>, Tensor<Scalar>)) {
     let value = conv2D(x, filter: filter, strides: strides, padding: padding, dilations: dilations)
     return (value, { v in
-        (conv2DBackpropInput(v, shape: x.shapeTensor, filter: filter,
-                             strides: strides, padding: padding, dilations: dilations),
+        (transposedConv2D(v, shape: x.shapeTensor, filter: filter,
+                          strides: strides, padding: padding, dilations: dilations),
          conv2DBackpropFilter(v, input: x, filterSizes: filter.shapeTensor,
                               strides: strides, padding: padding, dilations: dilations))
     })
 }
 
-/// TensorFlow builtin conv2d gradient helper for the input.
-@differentiable(wrt: (x, filter))
-@usableFromInline
-func conv2DBackpropInput<Scalar: TensorFlowFloatingPoint>(
-    _ x: Tensor<Scalar>,
+/// Returns a 2-D transposed convolution with the specified input, filter, strides, and padding.
+///
+/// - Parameters:
+///   - input: The input.
+///   - filter: The convolution filter.
+///   - strides: The strides of the sliding filter for each dimension of the input.
+///   - padding: The padding for the operation
+///   - dilations: The dilation factor for each dimension of the input.
+/// - Precondition: `input` must have rank `4`.
+/// - Precondition: `filter` must have rank 4.
+@differentiable(wrt: (input, filter))
+public func transposedConv2D<Scalar: TensorFlowFloatingPoint>(
+    _ input: Tensor<Scalar>,
     shape: Tensor<Int32>,
     filter: Tensor<Scalar>,
     strides: (Int, Int, Int, Int) = (1, 1, 1, 1),
     padding: Padding = .valid,
     dilations: (Int, Int, Int, Int) = (1, 1, 1, 1)
 ) -> Tensor<Scalar> {
+    precondition(input.shape.rank == 4, "The input must have rank 4.")
+    precondition(filter.shape.rank == 4, "The filter must have rank 4.")
     return _Raw.conv2DBackpropInput(
         inputSizes: shape,
         filter: filter,
-        outBackprop: x,
+        outBackprop: input,
```

Comment:
Do tests pass after this change? Afaik all convolution operations are checked with their gradients as well.

Reply:
It's merely a renaming of argument name.
I thought it should have same signature as `conv2D`.
Of course tests are passing.

Reply:
@t-ae oh sorry, overlooked the renaming!